### DISCUSSION TITLE
Workflow runs on `issues` trigger but not on `schedule` trigger

### DISCUSSION THREAD

**Discussion by DilumAluthge** [community/community#26161] [Actions]
<p>The “on” section of my workflow file looks like this:</p>
<pre><code>on:
  schedule:
    - cron: '00 * * * *'
  issues:
    types: [opened, reopened]
</code></pre>
<p>The workflow gets run on the “issues” trigger, but does not get run on the “schedule” trigger, as you can see from looking at the history here: <a href="https://github.com/bcbi/CompatHelper.jl/actions" rel="nofollow noopener">https://github.com/bcbi/CompatHelper.jl/actions</a></p>
<p>My full workflow file is here: <a href="https://github.com/bcbi/CompatHelper.jl/blob/master/.github/workflows/CompatHelper.yml" rel="nofollow noopener">https://github.com/bcbi/CompatHelper.jl/blob/master/.github/workflows/CompatHelper.yml</a></p>
<p>Now here is the weird part. There is a different repository that has the same exact workflow file, but the workflow runs correctly on both the <code>issues</code> and <code>schedule</code> triggers in that repository: <a href="https://github.com/JuliaImages/Images.jl/actions" rel="nofollow noopener">https://github.com/JuliaImages/Images.jl/actions</a></p>
<p>Here is the workflow file for that repository: <a href="https://github.com/JuliaImages/Images.jl/blob/master/.github/workflows/CompatHelper.yml" rel="nofollow noopener">https://github.com/JuliaImages/Images.jl/blob/master/.github/workflows/CompatHelper.yml</a></p>
<p>In particular, notice that the workflow file in the Images.jl repository has the following “on” section:</p>
<pre><code>on:
  schedule:
    - cron: '00 * * * *'
  issues:
    types: [opened, reopened]
</code></pre>

**Comment by wipe2238** [comment]
<pre><code>- cron: 0 * * * *
</code></pre>

**Comment by DilumAluthge** [comment]
<p>Well, here is the weird part. There is a different repository that has the same exact workflow file, but the workflow runs correctly on both the <code>issues</code> and <code>schedule</code> triggers in that repository: <a href="https://github.com/JuliaImages/Images.jl/actions" rel="nofollow noopener">https://github.com/JuliaImages/Images.jl/actions</a></p>
<p>Here is the workflow file for that repository: <a href="https://github.com/JuliaImages/Images.jl/blob/master/.github/workflows/CompatHelper.yml" rel="nofollow noopener">https://github.com/JuliaImages/Images.jl/blob/master/.github/workflows/CompatHelper.yml</a></p>
<p>In particular, notice that the Images.jl repository has the following “on” section:</p>
<pre><code>on:
  schedule:
    - cron: '00 * * * *'
  issues:
    types: [opened, reopened]
</code></pre>

**Comment by SaschaMann** [comment]
<p>I tried to reproduce it in a fork. Works just fine there: <a href="https://github.com/SaschaMann/CompatHelper.jl/actions" rel="nofollow noopener">https://github.com/SaschaMann/CompatHelper.jl/actions</a></p>

**Comment by wipe2238** [comment]
<p>You’re right, this is weird indeed :slight_smile:</p>
<p>Only difference i can see here is that in <code>JuliaImages</code> workflow <a href="https://github.com/JuliaImages/Images.jl/commits/master/.github/workflows/CompatHelper.yml" rel="nofollow noopener">never changed</a> since it was created with ‘00 …’, while <code>CompatHelper.jl</code> have multiple tweaks/edits to it. And hey, <code>CompatHelper.jl</code> schedule worked for a <a href="https://github.com/bcbi/CompatHelper.jl/actions?query=event%3Aschedule" rel="nofollow noopener">very short time</a> but then broke <a href="https://github.com/bcbi/CompatHelper.jl/commit/1a80d705a8109bb7c1ddc568acafef2ee4176a32#diff-ac0c48f0f5cc4292a0501cae75537f76" rel="nofollow noopener">after this commit</a> to never be seen again.</p>
<p>So maybe problem is behind scenes, and there’s a little tiny difference in how cron settings are <em>updated</em> vs how they’re <em>created</em> when using bit weird-looking syntax? That’s a wild guessing, obviously, there’s no way for both of us to check if that’s correct or i’m just talking nonsense here :confused:</p>

**Comment by DilumAluthge** [comment, marked as answer]
<p>I deleted the workflow and recreated it, and now the workflow is correctly running on the cron schedule: <a href="https://github.com/bcbi/CompatHelper.jl/actions" rel="nofollow noopener">https://github.com/bcbi/CompatHelper.jl/actions</a></p>
<p>I am still not sure why it wasn’t working originally. I suspect it might have something to do with the merge-bot that we use (Bors), but I don’t have any evidence of this.</p>

**Comment by deivid-rodriguez** [comment]
<p>Hi! We have a similar issue at <a href="https://github.com/rubygems/rubygems" rel="nofollow noopener">https://github.com/rubygems/rubygems</a>. I setup a scheduled workflow: <a href="https://github.com/rubygems/rubygems/blob/master/.github/workflows/daily.yml" rel="nofollow noopener">https://github.com/rubygems/rubygems/blob/master/.github/workflows/daily.yml</a>, but I never got it to run.</p>
<p>And… we do use Bors too… So it could indeed be related.</p>
<p><span class="mention">@dilumaluthgeWhat</span> did you do exactly to fix this? Thanks!</p>

**Comment by DilumAluthge** [comment]
<p>Here is what I did.</p>
<p>Step 1: Make a PR that deletes all workflows from the `.github/workflows/ directory. Merge this PR manually, i.e. by clicking the big green “Merge” button on the GitHub website. Do NOT merge this PR with Bors.</p>
<p>Step 2: Make a PR that adds all the workflows you want to have. As before, merge this PR manually, i.e. by clicking the big green “Merge” button on the GitHub website. Do NOT merge this PR with Bors.</p>
<p>Now your workflows should work correctly.</p>
<p>In the future, if you ever need to make a PR that modifies any of the workflow files in <code>.github/workflows</code>, make sure that you merge those PRs manually. If you try to use Bors to merge a PR that modifies a GitHub Actions workflow file, then things will break.</p>

**Comment by deivid-rodriguez** [comment]
<p>Thank you!</p>
<p>We were indeed having issues with bors when a PR coming from an external fork modified github actions files: <a href="https://github.com/bors-ng/bors-ng/issues/806" rel="nofollow noopener">https://github.com/bors-ng/bors-ng/issues/806</a>.</p>
<p>Manually merging this PR actually fixed the scheduled workflows issue as you suggested! :slight_smile: We’ll keep this in mind for the future.</p>
